### ISSUE TITLE
feat(autodiff): nn::Module に train／eval モードと named_parameters を追加する

### ISSUE THREAD

**Issue by aLiz-Nancy** [Fandhe-AI/fandhe-ai#1758]
## 概要
親 #1617「feat(autodiff): nn::Module の train／eval モード・named_parameters・ModuleList／汎用 Sequential コンテナを追加する」の 2h 粒度分解（2026-09-14）。`nn::Module` に train/eval モード・`named_parameters` を追加する。Dropout（#1603）・BatchNorm（#1608）の前提となる。
見積: 約 2h

## 受け入れ条件
- [ ] 対応する Op／`BackendOps` メソッド・Var メソッドを追加する（該当バックエンドすべてに実装する）
- [ ] VJP を追加し、既存 autodiff テストに数値微分または解析的検証を追加する
- [ ] parity テスト（REQ-2 複合判定、または bit 同一が要求される場合はそれ）を追加する
- [ ] facade 公開面に追加する場合は `docs/compat-api-scope.md` の対象範囲記述を更新する
- [ ] `cargo fmt --all`・`cargo clippy --workspace --all-targets --all-features -- -D warnings` を通す

## 契約・注意
- tolerance・baseline（許容誤差・数値一致ベースライン）の変更は本 issue の対象外。変更が必要になった場合は別途ユーザー承認を得る

## 承認事項
実装着手前に、facade 公開面（`fandhe_ai`／`compat`）の拡張について `docs/compat-api-scope.md` §5 の範囲拡張手続きに基づくユーザー承認が必須。本 issue の起票は承認ではない。前提の spec 改定（前提: #1591）が未承認の間は着手できない。

## 依存
依存: なし

## 出典
- 親 issue: #1617（feat(autodiff): nn::Module の train／eval モード・named_parameters・ModuleList／汎用 Sequential コンテナを追加する）
- 低レイヤー診断 artifact §7（https://claude.ai/code/artifact/4e107064-a190-4861-897d-3dce44d05428）
- 敗因分析 artifact（https://claude.ai/code/artifact/bce4801e-cc6b-4de9-a930-5eb8cf2ea15e）
- 機能ギャップ表: `docs/compat-feature-gap.md`（scratchpad `analysis/feature-gap.md` §2 を取り込み予定）
- 機能ギャップ表 §2.7、§2.15 行（`scratchpad/analysis/feature-gap.md`）
